include/torch-mlir/Conversion/TorchToTosa/TosaLegalizeCommon.h

```diff
@@ -21,6 +23,21 @@ createOneDimTfIndices(PatternRewriter &rewriter, Operation *op,
                       SmallVector<int64_t> indiceOneDimShape, int32_t dim,
                       ArrayRef<int64_t> indexShape);
 
+std::optional<Value> createMulOpAndCast(PatternRewriter &rewriter,
+                                        Operation *op, TensorType outType,
+                                        Value lhs, Value rhs, int32_t shift);
+
+// Create TOSA elementwise binary op with type conversion if necessary.
+template <typename TosaOpT>
+std::optional<Value> createBinaryOpAndCast(PatternRewriter &rewriter,
```

Comment:
why does this return `std::optional<Value>` instead of a stronger `TosaOpT`? `CreateOpAndInfer` also returns `TosaOpT`

Reply:
Agree with you. These are createOp functions that intend to return an operation but not conversion-kind functions that transfer a complicated operation into a sequence of tosa ops. I think the latter focus more on the result (value).

Change to return tosa operation.